client/my-sites/themes/themes-banner/random-themes-banner.jsx

```diff
@@ -21,13 +17,29 @@ class RandomThemesBanner extends PureComponent {
 		const { banners } = props;
 		this.state = {
 			BannerComponent: sample( banners ),
+			isClient: false,
 		};
 	}
 
+	componentDidMount() {
+		// In this particular case, we need to disable the ESLint check
+		// to disable using `setState()` on `componentDidMount()`, since
+		// it needs to be done this way to properly render the component
+		// both on client-side and server-side.
+		const state = this.state;
+
+		// eslint-disable-next-line react/no-did-mount-set-state
+		this.setState( {
+			...state,
```

Comment:
The `...state` bit isn't needed here, `setState()` is smart enough to preserve any other component state (tho in this case, we don't even have any).